catalog/spatial/registry/registry-schema-bindings/src/main/java/org/codice/ddf/registry/schemabindings/converter/type/AbstractRegistryObjectTypeConverter.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.registry.schemabindings.converter.type;
+
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.CLASSIFICATION_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.DESCRIPTION_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.EXTERNAL_IDENTIFIER_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.HOME_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.ID_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.LID_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.NAME_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.OBJECT_TYPE_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.SLOT;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.STATUS_KEY;
+import static org.codice.ddf.registry.schemabindings.converter.web.RegistryObjectWebConverter.VERSION_INFO_KEY;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.codice.ddf.registry.schemabindings.helper.MapToSchemaElement;
+
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ClassificationType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.ExternalIdentifierType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.SlotType1;
+
+public abstract class AbstractRegistryObjectTypeConverter<T extends RegistryObjectType> {
+
+    protected abstract MapToSchemaElement<T> getSchemaMapper();
```

Comment:
Any reason we force sub-classes to implement this method instead of taking it in as an argument of this abstract class' constructor?